src/framework/components/light/component.js

```diff
@@ -89,36 +89,140 @@ pc.extend(pc, function () {
      * @extends pc.Component
      */
 
+    var _props = [];
+    var _propsDefault = [];
+    function _defineProperty(name, defaultValue, setFunc) {
+
+        var c = LightComponent.prototype;
+        _props.push(name);
+        _propsDefault.push(defaultValue);
+
+        Object.defineProperty(c, name, {
+            get: function () {
+                return this.data[name];
+            },
+            set: function (value) {
+                var data = this.data;
+                var oldValue = data[name];
+                data[name] = value;
+                if (setFunc) setFunc(this, value, oldValue)
```

Comment:
`setFunc.call(this, value, oldValue)` - will make sure context is passed. So no "self" argument required in actual function later.